### ISSUE TITLE
Accessibility: Grouped buttons have incorrect role and missing field set – radio

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51918]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

This element is a collection of links that function as radio (single select) inputs.  Additionally, they are not grouped in a fieldset, and thus, the label for the group of buttons is not announced by screen readers when encountering the first one. Screen reader users will lack the broader context of what they are selecting without hearing the group label.

(There is a label above the stars, but it's associated with a select field set to display:none, which is hidden from screen readers, so the label appears orphaned.)

![Image](https://github.com/user-attachments/assets/7ad87699-6320-467a-8fc0-309275f5d7ff)

### Component or Page: 

Page: Product Single (Simple)  

### Name affected element:

Your Rating links
- 1 link, 2 link, 3 link, 4 link, 5 link

### Relevant code snippets:

`<span><a class=""star-1"" href=""#"">1</a><a class=""star-2"" href=""#"">2</a><a class=""star-3"" href=""#"">3</a>							<a class=""star-4"" href=""#"">4</a><a class=""star-5"" href=""#"">5</a></span>`

### Recommended fix:

"Add the role=""radio"" to indicate that only one rating may be selected at a time and use aria-checked=""true"" to indicate which of the ratings has been selected.
 
 For expected keyboard navigation that emulates a radio field, the links that are not currently selected should have tabindex=""-1"" so that only the currently selected button can be reached via the tab key. JavaScript keyboard handlers should be added so users can use the left and right Arrow keys to move through the links as they would be able to do with radio inputs.
 
 Wrap the entire component in a fieldset tag and use a legend tag to connect the label with the buttons instead of a label tag, as label tags are intended to describe a single field.
 
Add aria-labels to the links or change the off-screen text to be more descriptive. Instead of "1" say "1 of 5 stars."

**Comment by Manussakis** [comment]
I have started looking into this issue.
